source/agora/consensus/data/Transaction.d

```diff
@@ -260,10 +253,28 @@ public struct Output
     /// Support for sorting
     public int opCmp (in typeof(this) rhs) const nothrow @safe @nogc
     {
+        if (this.type != rhs.type)
+            return this.type < rhs.type ? -1 : 1;
         if (this.lock != rhs.lock)
             return this.lock < rhs.lock ? -1 : 1;
         return this.value.opCmp(rhs.value);
     }
+
+    /***************************************************************************
+
+        Transactions Serialization
+
+        Params:
+            dg = Serialize function
+
+    ***************************************************************************/
+
+    public void serialize (scope SerializeDg dg) const @trusted
+    {
+        serializePart(this.value, dg);
+        serializePart(this.lock, dg);
+        serializePart(this.type, dg);
+    }
```

Comment:
Why ?

Reply:
Removing this and re-ordering the fields in the Transaction. Also changing enum from `ubyte` to `unit`.